diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/BlockStructureBraces.kt

```diff
@@ -0,0 +1,118 @@
+import com.pinterest.ktlint.core.ast.ElementType.WHITE_SPACE
+import org.cqfn.diktat.common.config.rules.RulesConfig
+import org.cqfn.diktat.ruleset.constants.Warnings.BRACES_BLOCK_STRUCTURE_ERROR
+import org.cqfn.diktat.ruleset.rules.files.FileSize
+import org.cqfn.diktat.ruleset.utils.findChildBefore
+import org.cqfn.diktat.ruleset.utils.getFirstChildWithType
+import org.cqfn.diktat.ruleset.utils.hasChildOfType
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+import org.jetbrains.kotlin.com.intellij.psi.tree.IElementType
+import org.jetbrains.kotlin.psi.KtIfExpression
+import org.slf4j.LoggerFactory
+
+class BlockStructureBraces : Rule("block-structure") {
+    companion object {
+        private val log = LoggerFactory.getLogger(FileSize::class.java)
+    }
+
+    private lateinit var configRules: List<RulesConfig>
+    private lateinit var emitWarn: ((offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit)
+    private var fileName: String? = null
+    private var isFixMode: Boolean = false
+
+    override fun visit(node: ASTNode,
+                       autoCorrect: Boolean,
+                       params: KtLint.Params,
+                       emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit) {
+        configRules = params.getDiktatConfigRules()
+        fileName = params.fileName
+        emitWarn = emit
+        isFixMode = autoCorrect
+
+        when (node.elementType) {
+            FUN -> checkFun(node)
+            IF -> checkIf(node)
+            WHEN -> checkWhen(node)
+            FOR -> checkFor(node)
+        }
+    }
+
+    private fun checkFor(node: ASTNode) {
+        checkOpenBrace(node, BODY)
+        checkCloseBrace(node.findChildByType(BODY)!!.findChildByType(BLOCK)!!)
+    }
+
+    private fun checkWhen(node: ASTNode) {
+        checkOpenBrace(node, LBRACE)
+        checkCloseBrace(node)
+    }
+
+    private fun checkFun(node: ASTNode) {
+        if (node.hasChildOfType(BLOCK)) {
+            checkOpenBrace(node, BLOCK)
+            checkCloseBrace(node.findChildByType(BLOCK)!!)
+        }
+    }
+
+    private fun checkIf(node: ASTNode) {
+        val ifPsi = node.psi as KtIfExpression
+        val thenNode = ifPsi.then?.node
+        val hasElseBranch = ifPsi.elseKeyword != null
+        val elseNode = ifPsi.`else`?.node
+
+        checkOpenBrace(node, THEN)
+        checkCloseBrace(thenNode!!)
+        if (hasElseBranch && elseNode!!.elementType != IF) {
+            checkOpenBrace(node, ELSE)
+            checkCloseBrace(elseNode)
+        }
+    }
+
+    private fun checkOpenBrace(node: ASTNode, beforeType: IElementType) {
+        val braceSpace = node.findChildBefore(beforeType, WHITE_SPACE)
+        if (!(braceSpace == null || braceSpace.elementType != WHITE_SPACE)) {
+            if (braceSpace.text.contains("\n".toRegex())) {
+                BRACES_BLOCK_STRUCTURE_ERROR.warnAndFix(configRules, emitWarn, isFixMode, "open braces", 0) {
```

Comment:
```suggestion
                BRACES_BLOCK_STRUCTURE_ERROR.warnAndFix(configRules, emitWarn, isFixMode, "incorrect newline before opening brace", 0) {
```
To make error message more useful for the user.